WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
+    }
+}
+
+- (void)purgeJetpackAccount:(WPAccount *)account
```

Comment:
The fact that this method would work on a regular blog now is purely coincidental.  There's no requirement for future modifications or additions to maintain this, which means the blog being a Jetpack one is indeed a precondition to this method (given it's current name and documentation).

Alternatively you can make this method generic (such as naming it `purgeAccount` and documenting it) so that future modifications make sure this still works for all types of blogs.